### ISSUE TITLE
Remote specific protocol settings (configuration, feature request)

### ISSUE THREAD

**Issue by mpacer** [github/hub#1435]
Would it be possible to have a default protocol specific to different commands?

Specifically the use case I have in mind is having ssh access to all of my personal repositories, but only https access to remote repositories. 

This acts as a safety measure, as I can easily push to my local repositories over ssh, but need to go through additional steps to push on remote repositories.

↬@Carreau for introducing this method to me.

**Comment by mislav** [comment]
Will this be addressed if we add a `.hubconfig` file that you can put in the working copy of an individual repository? Because that is planned.

**Comment by mpacer** [comment]
Not quite. This would involve being able to specify a ssh vs https connection for different remotes on the same repository, specifically upstream vs. origin. In practice they are both remote they are just associated with my github repo (e.g., https://github.com/mpacer/nbconvert) vs. the upstream (https://github.com/jupyter/nbconvert). 

Failure on my part to use language precisely originally.

Local and remote should have been origin and upstream, changing that now.

**Comment by mislav** [comment]
Ah, I see now.

Where do you envision these configuration options be kept? Would this global configuration encompass all same-named remotes, e.g. special casing for "upstream" remote (if configured) would be applied to every repo on your machine?

**Comment by Carreau** [comment]
I guess using something like [conditional includes](https://git-scm.com/docs/git-config#_conditional_includes), and/or [pushInsteadOf](https://git-scm.com/docs/git-config#git-config-urlltbasegtpushInsteadOf), though these two options do now allow to match on the name of the remote but only on the urls/gitdir. I don't know of any config that allow a per-remote-name configuration but I at least tend to name my remote consistently so a global config would indeed be awesome.
